.github/workflows/e2ePerformanceTests.yml

```diff
@@ -74,15 +72,17 @@ jobs:
         if: ${{ fromJSON(steps.getPullRequestDetails.outputs.IS_MERGED) }}
         id: getMergeCommitShaIfMergedPR
         run: |
-          echo "MERGE_COMMIT_SHA=${{ steps.getPullRequestDetails.outputs.MERGE_COMMIT_SHA }}" >> "$GITHUB_OUTPUT"
+          MERGE_COMMIT_SHA=${{ steps.getPullRequestDetails.outputs.MERGE_COMMIT_SHA }}
+          git fetch origin "$MERGE_COMMIT_SHA" --no-tags --depth=1
```

Comment:
Well first of all, this workflow is written in kind of a confusing way imo, but I didn't want to make more changes than necessary.

- I removed the `fetch-depth: 0` from above, so that means any ref other than `github.ref` (`main` branch in this case) will need to be fetched
- Lines 71-79 get the `MERGE_COMMIT_SHA` if the PR was already merged
- Lines 93-99 get the `MERGE_COMMIT_SHA` if the PR is not yet merged. It does this by locally merging the `HEAD_COMMIT_SHA` into the main branch w/o creating a commit. Then outputs the new `HEAD` as `MERGE_COMMIT_SHA`.
    - Note: you helped me find a bug here because I needed to also fetch the `HEAD_COMMIT_SHA` in this step first. Good job 👏🏼 
- Lines 105-109 coalesce those `MERGE_COMMIT_SHA`s and get one `DELTA_REF`
- Lines 111-112 checkout the `DELTA_REF` (which means that it needs to be fetched first)

Reply:
**Edit:** Actually my first fix was incorrect ... the `HEAD_COMMIT_SHA` of an unmerged PR was incorrect. That commit was fetched (for PRs coming from a fork) in the step that is now called "Unmerged PR - Fetch head ref of unmerged PR".

Before, that step would only run for forked PRs. I've updated such that it will always run, and will fetch the commit from the correct remote.

Reply:
Still, there was a 🐛 so good question 👏🏼

Reply:
Glad it helped
![image](https://github.com/Expensify/App/assets/36083550/476a5f98-f643-42d0-8139-9d34d402c350)